gse-afs-ext-base/src/main/java/com/powsybl/gse/afs/ext/base/ImportedCaseCreator.java

```diff
@@ -163,4 +176,43 @@ public BooleanBinding okProperty() {
     @Override
     public void dispose() {
     }
+
+    private void replaceNode(ProjectFolder folder, Case aCase, Scene scene) {
+        String name = aCase.getName();
+        Optional<ButtonType> result = GseAlerts.showReplaceAndQuitDialog(folder.getName(), name);
+        result.ifPresent(buttonType -> {
+            if (buttonType.getButtonData() == ButtonBar.ButtonData.YES) {
+                folder.getChild(name).ifPresent(projectNode -> {
+                    backwardDependencies = projectNode.getBackwardDependencies();
+                    projectNode.rename(TEMPORARY_NAME);
+                    buildFile(aCase, folder, null);
+                    folder.getChild(name).ifPresent(importedNode ->
```

Comment:
if the node creation fail (folder.getChild return an empty optional) revert the name of the original node